### PR TITLE
added blog article for cost of testing pipelines

### PR DESCRIPTION
**Pull request by dan-mm**

### CODE REVIEW

**Comment by ArneTR** [comment]
Original discussion via e-mail:

> Pushed my latest version. some notes / what's unfinished:

> 1 - I originally wanted to rewrite this with suggestions on how to save energy via good practices in reducing workflows, using moby as a prime example, but when I looked more in depth at how moby was running their workflows, it already was pretty efficient. I didn't see a way to suggest any significant savings, so I went back to the original plan of the point of the article to just be getting a tangible feel for what a CI pipeline's output is in terms of real world carbon cost. Hopefully the story thread for this is better now.

> 2- after crunching the numbers on moby and writing that section - I realize there was a mistake in my base assumption. I calculated a mJ/s estimation based on our measured workflows, and then found the total time run for all moby workflows and multiplied by this number. I thought this was safe because Github only uses one type of linux machine now for public runner. However, two of their longest running workflows run on windows machines, not linux, this base mJ/s number I used not really applicable :-/ 

> Unfortunately we don't have data on windows machines at all to get a better estimation. We should either remove the moby section entirely (maybe look at curl repo as a whole instead?) or just point this out. Whatever the windows mJ/s would be is different, however order of magnitude-wise it would still be in the same ballpark. The whole thing is an estimation anyways.

> Still left to do:
> - Table formatting for moby section / prettifying
> - conclusion (atm its just two sentences as to what my intention for the concluding paragraph is)

My remarks: 


I really like the new version of the article. 

2. Also nice to see that the moby part is in there. I am a bit confused that the numbers are so low. Aren't the usage amounts way higher than curl?

3. The Windows thing should definitely be highlighted and maybe even be excluded.

4. I think bringing in optimization tips into this article is a bit heavy anyway. But a teaser to the next article would be nice.

**Comment by ArneTR** [comment]
@ribalba Just overhauled the article. Please give it a look

File: content/case-studies/carbon-cost-of-testing-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+windows-2022.yml - 467 runs - 17100 seconds each - 7985700 total
+
+Adding these all together and multiplying by our mJ/s gives us an estimation of 95,350,445,093.4 mJ of energy used. Converting then to gCO2 the same way as above we get: 11706,92 gCO2 emitted, for entire whole month by all workflows. And now to put it in some real world context:
+
+11706,92 gCO2
+30 miles driven
+1.3 gallons of gasoline consumed
+1,424 smartphones charged
+
+While these numbers aren't exactly scary, they are starting to feel tangible. And this is one repository, working for one month - when looking at the bigger picture we start to see how this can scale up. <outro - more here>
```
Comment (ribalba) [inline content/case-studies/carbon-cost-of-testing-pipelines.md:294]: outo - more here?

**Review by ribalba — APPROVED** [review]
one minor thing :) 

maybe we need a ChatGPT commit hook that you can use
```
<ChatGPT prompt="outro here"> 
```

otherwise a nice write up with good numbers

**Pull request merged**